In [2]:
import pandas as pd
import numpy as np

Abaixo estou importando o .csv que vamos trabalhar e como o cabeçalho esta na linha 0 eu coloco o header como 0 assim não preciso definir as colunas:

In [7]:
df=pd.read_csv('./data/Video_Games.csv',header=0)
df.head(10)

Name Platform  Year_of_Release         Genre  \
0                 Wii Sports      Wii           2006.0        Sports   
1          Super Mario Bros.      NES           1985.0      Platform   
2             Mario Kart Wii      Wii           2008.0        Racing   
3          Wii Sports Resort      Wii           2009.0        Sports   
4   Pokemon Red/Pokemon Blue       GB           1996.0  Role-Playing   
5                     Tetris       GB           1989.0        Puzzle   
6      New Super Mario Bros.       DS           2006.0      Platform   
7                   Wii Play      Wii           2006.0          Misc   
8  New Super Mario Bros. Wii      Wii           2009.0      Platform   
9                  Duck Hunt      NES           1984.0       Shooter   

  Publisher  NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  \
0  Nintendo     41.36     28.96      3.77         8.45         82.53   
1  Nintendo     29.08      3.58      6.81         0.77         40.24   
2  Nintendo     15.68     12.76      3.79         3.29         35.52   
3  Nintendo     15.61     10.93      3.28         2.95         32.77   
4  Nintendo     11.27      8.89     10.22         1.00         31.37   
5  Nintendo     23.20      2.26      4.22         0.58         30.26   
6  Nintendo     11.28      9.14      6.50         2.88         29.80   
7  Nintendo     13.96      9.18      2.93         2.84         28.92   
8  Nintendo     14.44      6.94      4.70         2.24         28.32   
9  Nintendo     26.93      0.63      0.28         0.47         28.31   

   Critic_Score  Critic_Count User_Score  User_Count Developer Rating  
0          76.0          51.0          8       322.0  Nintendo      E  
1           NaN           NaN        NaN         NaN       NaN    NaN  
2          82.0          73.0        8.3       709.0  Nintendo      E  
3          80.0          73.0          8       192.0  Nintendo      E  
4           NaN           NaN        NaN         NaN       NaN    NaN  
5           NaN           NaN        NaN         NaN       NaN    NaN  
6          89.0          65.0        8.5       431.0  Nintendo      E  
7          58.0          41.0        6.6       129.0  Nintendo      E  
8          87.0          80.0        8.4       594.0  Nintendo      E  
9           NaN           NaN        NaN         NaN       NaN    NaN

In [9]:
display(df)

Name Platform  Year_of_Release         Genre  \
0                         Wii Sports      Wii           2006.0        Sports   
1                  Super Mario Bros.      NES           1985.0      Platform   
2                     Mario Kart Wii      Wii           2008.0        Racing   
3                  Wii Sports Resort      Wii           2009.0        Sports   
4           Pokemon Red/Pokemon Blue       GB           1996.0  Role-Playing   
...                              ...      ...              ...           ...   
16714  Samurai Warriors: Sanada Maru      PS3           2016.0        Action   
16715               LMA Manager 2007     X360           2006.0        Sports   
16716        Haitaka no Psychedelica      PSV           2016.0     Adventure   
16717               Spirits & Spells      GBA           2003.0      Platform   
16718            Winning Post 8 2016      PSV           2016.0    Simulation   

          Publisher  NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  \
0          Nintendo     41.36     28.96      3.77         8.45         82.53   
1          Nintendo     29.08      3.58      6.81         0.77         40.24   
2          Nintendo     15.68     12.76      3.79         3.29         35.52   
3          Nintendo     15.61     10.93      3.28         2.95         32.77   
4          Nintendo     11.27      8.89     10.22         1.00         31.37   
...             ...       ...       ...       ...          ...           ...   
16714    Tecmo Koei      0.00      0.00      0.01         0.00          0.01   
16715   Codemasters      0.00      0.01      0.00         0.00          0.01   
16716  Idea Factory      0.00      0.00      0.01         0.00          0.01   
16717       Wanadoo      0.01      0.00      0.00         0.00          0.01   
16718    Tecmo Koei      0.00      0.00      0.01         0.00          0.01   

       Critic_Score  Critic_Count User_Score  User_Count Developer Rating  
0              76.0          51.0          8       322.0  Nintendo      E  
1               NaN           NaN        NaN         NaN       NaN    NaN  
2              82.0          73.0        8.3       709.0  Nintendo      E  
3              80.0          73.0          8       192.0  Nintendo      E  
4               NaN           NaN        NaN         NaN       NaN    NaN  
...             ...           ...        ...         ...       ...    ...  
16714           NaN           NaN        NaN         NaN       NaN    NaN  
16715           NaN           NaN        NaN         NaN       NaN    NaN  
16716           NaN           NaN        NaN         NaN       NaN    NaN  
16717           NaN           NaN        NaN         NaN       NaN    NaN  
16718           NaN           NaN        NaN         NaN       NaN    NaN  

[16719 rows x 16 columns]

O arquivo .csv trás as vendas de jogos de video game desde os anos 80 até o anos 2010, boscamos analizar os totais de vendas, mais as notas de critica e as notas dos usuários, podendo analizar quais as plataformas e generos mais pupulares.

In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16719 entries, 0 to 16718
Data columns (total 16 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Name             16717 non-null  object 
 1   Platform         16719 non-null  object 
 2   Year_of_Release  16450 non-null  float64
 3   Genre            16717 non-null  object 
 4   Publisher        16665 non-null  object 
 5   NA_Sales         16719 non-null  float64
 6   EU_Sales         16719 non-null  float64
 7   JP_Sales         16719 non-null  float64
 8   Other_Sales      16719 non-null  float64
 9   Global_Sales     16719 non-null  float64
 10  Critic_Score     8137 non-null   float64
 11  Critic_Count     8137 non-null   float64
 12  User_Score       10015 non-null  object 
 13  User_Count       7590 non-null   float64
 14  Developer        10096 non-null  object 
 15  Rating           9950 non-null   object 
dtypes: float64(9), object(7)
memory usage: 2.0+ MB


Abaixo vamos eliminar as colunas 'NA_Sales','EU_Sales','JP_Sales','Other_Sales','Critic_Count','User_Count', 'Rating' já que vendas separadas não é interessante nesse momento e nem os numeros de criticos e usuários bem como a classificação dos jogos. 

In [10]:
df = df.drop(['NA_Sales','EU_Sales','JP_Sales','Other_Sales','Critic_Count','User_Count', 'Rating'], axis=1)

In [12]:
display(df)

Name Platform  Year_of_Release         Genre  \
0                         Wii Sports      Wii           2006.0        Sports   
1                  Super Mario Bros.      NES           1985.0      Platform   
2                     Mario Kart Wii      Wii           2008.0        Racing   
3                  Wii Sports Resort      Wii           2009.0        Sports   
4           Pokemon Red/Pokemon Blue       GB           1996.0  Role-Playing   
...                              ...      ...              ...           ...   
16714  Samurai Warriors: Sanada Maru      PS3           2016.0        Action   
16715               LMA Manager 2007     X360           2006.0        Sports   
16716        Haitaka no Psychedelica      PSV           2016.0     Adventure   
16717               Spirits & Spells      GBA           2003.0      Platform   
16718            Winning Post 8 2016      PSV           2016.0    Simulation   

          Publisher  Global_Sales  Critic_Score User_Score Developer  
0          Nintendo         82.53          76.0          8  Nintendo  
1          Nintendo         40.24           NaN        NaN       NaN  
2          Nintendo         35.52          82.0        8.3  Nintendo  
3          Nintendo         32.77          80.0          8  Nintendo  
4          Nintendo         31.37           NaN        NaN       NaN  
...             ...           ...           ...        ...       ...  
16714    Tecmo Koei          0.01           NaN        NaN       NaN  
16715   Codemasters          0.01           NaN        NaN       NaN  
16716  Idea Factory          0.01           NaN        NaN       NaN  
16717       Wanadoo          0.01           NaN        NaN       NaN  
16718    Tecmo Koei          0.01           NaN        NaN       NaN  

[16719 rows x 9 columns]

Verificando valores únicos, dessa forma podemos diferenciar as quantidades de plataformas e generos como outras informações: 

In [13]:
df.nunique()

Name               11562
Platform              31
Year_of_Release       39
Genre                 12
Publisher            581
Global_Sales         629
Critic_Score          82
User_Score            96
Developer           1696
dtype: int64

Abaixos podemos ver que a principal plataforma é o Playstation 2 (PS2 seguido do Nintendo DS (DS)).

In [14]:
df['Platform'].value_counts()

Platform
PS2     2161
DS      2152
PS3     1331
Wii     1320
X360    1262
PSP     1209
PS      1197
PC       974
XB       824
GBA      822
GC       556
3DS      520
PSV      432
PS4      393
N64      319
XOne     247
SNES     239
SAT      173
WiiU     147
2600     133
NES       98
GB        98
DC        52
GEN       29
NG        12
SCD        6
WS         6
3DO        3
TG16       2
GG         1
PCFX       1
Name: count, dtype: int64

Verificando as principais Publisher que são as empresas que publicam os jogos mas não necessáriamente o desenvolvem.

In [15]:
df['Publisher'].value_counts()

Publisher
Electronic Arts                 1356
Activision                       985
Namco Bandai Games               939
Ubisoft                          933
Konami Digital Entertainment     834
                                ... 
TalonSoft                          1
RED Entertainment                  1
Red Flagship                       1
Westwood Studios                   1
Ongakukan                          1
Name: count, Length: 581, dtype: int64

Verificando as principais Desenvolvedoras.

In [16]:
df['Developer'].value_counts()

Developer
Ubisoft          204
EA Sports        172
EA Canada        167
Konami           162
Capcom           139
                ... 
Logris Games       1
Boomzap            1
ChaYoWo Games      1
Inferno Games      1
Papyrus            1
Name: count, Length: 1696, dtype: int64

Abaixo vou remover as linhas onde nas colunas (Name, Platform, Year_of_Release, Genre, Global_Sales) tenha algum dado ausente: 

In [18]:
df = df.dropna(subset=['Name', 'Platform', 'Year_of_Release', 'Genre', 'Global_Sales'])

In [19]:
display(df)

Name Platform  Year_of_Release         Genre  \
0                         Wii Sports      Wii           2006.0        Sports   
1                  Super Mario Bros.      NES           1985.0      Platform   
2                     Mario Kart Wii      Wii           2008.0        Racing   
3                  Wii Sports Resort      Wii           2009.0        Sports   
4           Pokemon Red/Pokemon Blue       GB           1996.0  Role-Playing   
...                              ...      ...              ...           ...   
16714  Samurai Warriors: Sanada Maru      PS3           2016.0        Action   
16715               LMA Manager 2007     X360           2006.0        Sports   
16716        Haitaka no Psychedelica      PSV           2016.0     Adventure   
16717               Spirits & Spells      GBA           2003.0      Platform   
16718            Winning Post 8 2016      PSV           2016.0    Simulation   

          Publisher  Global_Sales  Critic_Score User_Score Developer  
0          Nintendo         82.53          76.0          8  Nintendo  
1          Nintendo         40.24           NaN        NaN       NaN  
2          Nintendo         35.52          82.0        8.3  Nintendo  
3          Nintendo         32.77          80.0          8  Nintendo  
4          Nintendo         31.37           NaN        NaN       NaN  
...             ...           ...           ...        ...       ...  
16714    Tecmo Koei          0.01           NaN        NaN       NaN  
16715   Codemasters          0.01           NaN        NaN       NaN  
16716  Idea Factory          0.01           NaN        NaN       NaN  
16717       Wanadoo          0.01           NaN        NaN       NaN  
16718    Tecmo Koei          0.01           NaN        NaN       NaN  

[16448 rows x 9 columns]

Como ainda existem muitas linhas no arquivos e dados para serem trabalhados vamos fazer a limpeza dos dados não encontrados nas colunas Critic_Score User_Score

In [20]:
df = df.dropna(subset=['Critic_Score', 'User_Score'])

In [21]:
display(df)

Name Platform  Year_of_Release      Genre  \
0                   Wii Sports      Wii           2006.0     Sports   
2               Mario Kart Wii      Wii           2008.0     Racing   
3            Wii Sports Resort      Wii           2009.0     Sports   
6        New Super Mario Bros.       DS           2006.0   Platform   
7                     Wii Play      Wii           2006.0       Misc   
...                        ...      ...              ...        ...   
16700                   Breach       PC           2011.0    Shooter   
16701         Bust-A-Move 3000       GC           2003.0     Puzzle   
16702         Mega Brain Boost       DS           2008.0     Puzzle   
16706  STORM: Frontline Nation       PC           2011.0   Strategy   
16709                  15 Days       PC           2009.0  Adventure   

                   Publisher  Global_Sales  Critic_Score User_Score  \
0                   Nintendo         82.53          76.0          8   
2                   Nintendo         35.52          82.0        8.3   
3                   Nintendo         32.77          80.0          8   
6                   Nintendo         29.80          89.0        8.5   
7                   Nintendo         28.92          58.0        6.6   
...                      ...           ...           ...        ...   
16700              Destineer          0.01          61.0        5.8   
16701                Ubisoft          0.01          53.0        tbd   
16702  Majesco Entertainment          0.01          48.0        tbd   
16706                Unknown          0.01          60.0        7.2   
16709      DTP Entertainment          0.01          63.0        5.8   

                Developer  
0                Nintendo  
2                Nintendo  
3                Nintendo  
6                Nintendo  
7                Nintendo  
...                   ...  
16700        Atomic Games  
16701   Taito Corporation  
16702  Interchannel-Holon  
16706              SimBin  
16709   DTP Entertainment  

[7950 rows x 9 columns]

Assim nosso arquivo ficou com apenas 7950 linhas:

Verificando novamente os valores únicos:

In [22]:
df.nunique()

Name               5011
Platform             17
Year_of_Release      25
Genre                12
Publisher           301
Global_Sales        537
Critic_Score         81
User_Score           90
Developer          1450
dtype: int64

In [23]:
df['Platform'].value_counts()

Platform
PS2     1275
X360     890
PS3      804
XB       705
DS       704
PC       691
Wii      566
PSP      455
GC       437
GBA      426
PS4      251
PS       189
XOne     168
3DS      166
PSV      119
WiiU      90
DC        14
Name: count, dtype: int64

In [24]:
df['Publisher'].value_counts()

Publisher
Electronic Arts                1014
Activision                      562
Ubisoft                         556
THQ                             402
Sony Computer Entertainment     348
                               ... 
FuRyu Corporation                 1
Victor Interactive                1
10TACLE Studios                   1
Codemasters Online                1
NewKidCo                          1
Name: count, Length: 301, dtype: int64

In [25]:
df['Developer'].value_counts()

Developer
EA Canada                      155
EA Sports                      149
Capcom                         131
Konami                         125
Ubisoft                        118
                              ... 
Aqua Plus                        1
Game Arts, Seta Corporation      1
Compile Heart, GCREST            1
Rockstar Vienna                  1
Buddiez, Inc.                    1
Name: count, Length: 1450, dtype: int64

Salvando o arquivo limpo:

In [26]:
df.to_csv('./data/Clear_Video_Games.csv' , index=False)